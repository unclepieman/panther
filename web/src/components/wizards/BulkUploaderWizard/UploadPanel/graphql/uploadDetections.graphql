mutation UploadDetections($input: UploadDetectionsInput!) {
  uploadDetections(input: $input) {
    totalPolicies
    modifiedPolicies
    newPolicies
    totalRules
    modifiedRules
    newRules
    totalGlobals
    newGlobals
    modifiedGlobals
    totalDataModels
    modifiedDataModels
    newDataModels
  }
}
