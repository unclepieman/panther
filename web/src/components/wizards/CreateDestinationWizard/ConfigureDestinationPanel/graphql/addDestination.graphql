mutation AddDestination($input: DestinationInput!) {
  addDestination(input: $input) {
    ...DestinationFull
  }
}
