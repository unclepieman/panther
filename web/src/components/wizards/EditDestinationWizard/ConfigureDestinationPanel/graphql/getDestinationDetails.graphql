query GetDestinationDetails($id: ID!) {
  destination(id: $id) {
    ...DestinationFull
  }
}
