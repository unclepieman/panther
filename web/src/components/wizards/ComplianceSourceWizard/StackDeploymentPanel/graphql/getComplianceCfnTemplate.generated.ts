/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import * as Types from '../../../../../../__generated__/schema';

import { GraphQLError } from 'graphql';
import gql from 'graphql-tag';
import * as ApolloReactCommon from '@apollo/client';
import * as ApolloReactHooks from '@apollo/client';

export type GetComplianceCfnTemplateVariables = {
  input: Types.GetComplianceIntegrationTemplateInput;
};

export type GetComplianceCfnTemplate = {
  getComplianceIntegrationTemplate: Pick<Types.IntegrationTemplate, 'body' | 'stackName'>;
};

export const GetComplianceCfnTemplateDocument = gql`
  query GetComplianceCfnTemplate($input: GetComplianceIntegrationTemplateInput!) {
    getComplianceIntegrationTemplate(input: $input) {
      body
      stackName
    }
  }
`;

/**
 * __useGetComplianceCfnTemplate__
 *
 * To run a query within a React component, call `useGetComplianceCfnTemplate` and pass it any options that fit your needs.
 * When your component renders, `useGetComplianceCfnTemplate` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetComplianceCfnTemplate({
 *   variables: {
 *      input: // value for 'input'
 *   },
 * });
 */
export function useGetComplianceCfnTemplate(
  baseOptions?: ApolloReactHooks.QueryHookOptions<
    GetComplianceCfnTemplate,
    GetComplianceCfnTemplateVariables
  >
) {
  return ApolloReactHooks.useQuery<GetComplianceCfnTemplate, GetComplianceCfnTemplateVariables>(
    GetComplianceCfnTemplateDocument,
    baseOptions
  );
}
export function useGetComplianceCfnTemplateLazyQuery(
  baseOptions?: ApolloReactHooks.LazyQueryHookOptions<
    GetComplianceCfnTemplate,
    GetComplianceCfnTemplateVariables
  >
) {
  return ApolloReactHooks.useLazyQuery<GetComplianceCfnTemplate, GetComplianceCfnTemplateVariables>(
    GetComplianceCfnTemplateDocument,
    baseOptions
  );
}
export type GetComplianceCfnTemplateHookResult = ReturnType<typeof useGetComplianceCfnTemplate>;
export type GetComplianceCfnTemplateLazyQueryHookResult = ReturnType<
  typeof useGetComplianceCfnTemplateLazyQuery
>;
export type GetComplianceCfnTemplateQueryResult = ApolloReactCommon.QueryResult<
  GetComplianceCfnTemplate,
  GetComplianceCfnTemplateVariables
>;
export function mockGetComplianceCfnTemplate({
  data,
  variables,
  errors,
}: {
  data: GetComplianceCfnTemplate;
  variables?: GetComplianceCfnTemplateVariables;
  errors?: GraphQLError[];
}) {
  return {
    request: { query: GetComplianceCfnTemplateDocument, variables },
    result: { data, errors },
  };
}
