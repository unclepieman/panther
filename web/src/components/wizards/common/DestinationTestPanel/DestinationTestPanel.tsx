/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { useWizardContext, WizardPanel } from 'Components/Wizard';
import {
  AbstractButton,
  Box,
  Button,
  Card,
  Flex,
  Grid,
  Img,
  Link,
  Text,
  useSnackbar,
} from 'pouncejs';
import { Link as RRLink } from 'react-router-dom';
import urls from 'Source/urls';
import SuccessStatus from 'Assets/statuses/success.svg';
import NotificationStatus from 'Assets/statuses/notification.svg';
import LinkButton from 'Components/buttons/LinkButton';
import { extractErrorMessage } from 'Helpers/utils';
import { DeliveryResponseFull } from 'Source/graphql/fragments/DeliveryResponseFull.generated';
import { useSendTestAlertLazyQuery } from 'Source/graphql/queries';
import { EventEnum, SrcEnum, trackError, TrackErrorEnum, trackEvent } from 'Helpers/analytics';
import { WizardData as CreateWizardData } from '../../CreateDestinationWizard';
import { WizardData as EditWizardData } from '../../EditDestinationWizard';

type DeliveryResponses = Array<DeliveryResponseFull>;
type RowProps = { field: string; value: string | number };

const Row: React.FC<RowProps> = ({ field, value }) => {
  return (
    <React.Fragment>
      <Box as="dt" my="auto">
        {field}
      </Box>
      <Text as="dd" fontWeight="bold">
        {value}
      </Text>
    </React.Fragment>
  );
};

const DestinationTestPanel: React.FC = () => {
  const [testResponses, setTestResponses] = React.useState<DeliveryResponses>([]);
  const {
    data: { destination },
    reset,
    goToPrevStep,
  } = useWizardContext<CreateWizardData & EditWizardData>();

  const { pushSnackbar } = useSnackbar();

  const [sendTestAlert, { loading }] = useSendTestAlertLazyQuery({
    fetchPolicy: 'network-only', // Don't use cache
    variables: {
      input: {
        outputIds: [destination.outputId],
      },
    },
    // Failed deliveries will also trigger onCompleted as we don't return exceptions
    onCompleted: data => setTestResponses(data.sendTestAlert),
    // This will be fired if there was a network issue or other unknown internal exception
    onError: error => {
      trackError({ event: TrackErrorEnum.FailedDestinationTest, src: SrcEnum.Destinations, ctx: destination.outputType}) // prettier-ignore
      pushSnackbar({
        variant: 'error',
        title:
          extractErrorMessage(error) || 'Failed to send a test alert to the given destination(s)',
      });
    },
  });

  const handleTestAlertClick = React.useCallback(() => {
    trackEvent({ event: EventEnum.TestedDestination, src: SrcEnum.Destinations, ctx: destination.outputType }) // prettier-ignore
    sendTestAlert();
  }, [destination]);

  // We are not expecting more than one response since we are sending one ID
  if (testResponses.length && testResponses[0].success === false) {
    trackEvent({ event: EventEnum.TestedDestinationFailure, src: SrcEnum.Destinations, ctx: destination.outputType }) // prettier-ignore
    const { dispatchedAt, message, outputId, statusCode, success } = testResponses[0];
    return (
      <Box maxWidth={700} mx="auto">
        <WizardPanel.Heading
          title="Testing your Destination"
          subtitle="Something went wrong and the destination you have configured did not receive the test alert. Please update your destination settings and try again."
        />
        <Flex direction="column" align="center" spacing={6} my={6}>
          <Card backgroundColor="pink-700" p={6}>
            <Grid
              as="dl"
              wordBreak="break-word"
              templateColumns="max-content 1fr"
              fontSize="medium"
              fontWeight="medium"
              gap={4}
            >
              <Row field="Dispatched at" value={dispatchedAt} />
              <Row field="Message" value={message} />
              <Row field="Output ID" value={outputId} />
              <Row field="Status Code" value={statusCode} />
              <Row field="Success" value={success.toString()} />
            </Grid>
          </Card>
          <Text>
            If you don{"'"}t feel like it right now, you can always change the configuration later
          </Text>
          <LinkButton
            to={urls.integrations.destinations.edit(destination.outputId)}
            onClick={goToPrevStep}
          >
            Back to Configuration
          </LinkButton>
          <Link as={RRLink} variant="discreet" to={urls.integrations.destinations.list()}>
            Skip Testing
          </Link>
        </Flex>
      </Box>
    );
  }

  if (testResponses.length && testResponses.every(response => response.success === true)) {
    trackEvent({ event: EventEnum.TestedDestinationSuccessfully, src: SrcEnum.Destinations, ctx: destination.outputType }) // prettier-ignore
    return (
      <Box maxWidth={700} mx="auto">
        <WizardPanel.Heading
          title="Testing your Destination"
          subtitle="Everything worked as planned and your destination received the triggered alert. You can always send additional test alerts from the destinations page."
        />
        <Flex direction="column" align="center" spacing={6} my={6}>
          <Img
            nativeWidth={120}
            nativeHeight={120}
            alt="Test Alert received"
            src={NotificationStatus}
          />
          <Text>Signed, sealed, and delivered. You are good to go!</Text>
          <LinkButton to={urls.integrations.destinations.list()}>Finish Setup</LinkButton>
          <Link as={AbstractButton} variant="discreet" onClick={reset}>
            Add Another
          </Link>
        </Flex>
      </Box>
    );
  }

  return (
    <Box maxWidth={700} mx="auto">
      <WizardPanel.Heading
        title="Everything looks good!"
        subtitle="Your destination was successfully added and you will receive alerts based on your configuration. You can always edit or delete this destination from the destinations page"
      />
      <Flex direction="column" align="center" spacing={6} my={6}>
        <Img nativeWidth={120} nativeHeight={120} alt="Success" src={SuccessStatus} />
        <Text mb={5}>Do you want to try it out by sending a test Alert?</Text>
        <Box>
          <Button loading={loading} disabled={loading} onClick={handleTestAlertClick}>
            Send Test Alert
          </Button>
        </Box>
        <Link as={RRLink} variant="discreet" to={urls.integrations.destinations.list()}>
          Finish Setup
        </Link>
      </Flex>
    </Box>
  );
};

export default DestinationTestPanel;
