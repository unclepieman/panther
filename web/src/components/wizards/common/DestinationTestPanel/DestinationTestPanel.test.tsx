/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import {
  render,
  buildDestination,
  buildSendTestAlertInput,
  buildDeliveryResponse,
  fireClickAndMouseEvents,
  waitFor,
} from 'test-utils';
import { Wizard, WizardPanel } from 'Components/Wizard';
import { mockSendTestAlert } from 'Source/graphql/queries';
import DestinationTestPanel from './index';

const destination = buildDestination();
describe('DestinationTestPanel', () => {
  it('should show a success message for successful test-alert delivery', async () => {
    const deliveryResponse = buildDeliveryResponse({ statusCode: 200, success: true });
    const mocks = [
      mockSendTestAlert({
        variables: {
          input: buildSendTestAlertInput({ outputIds: [destination.outputId] }),
        },
        data: { sendTestAlert: [deliveryResponse] },
      }),
    ];
    const { getByText, container, getByAltText } = render(
      <Wizard initialData={{ destination }}>
        <Wizard.Step title="title">
          <WizardPanel>
            <DestinationTestPanel />
          </WizardPanel>
        </Wizard.Step>
      </Wizard>,
      { mocks }
    );

    expect(getByText('Everything looks good!')).toBeInTheDocument();
    const sendTestBtn = getByText('Send Test Alert');
    expect(container).toMatchSnapshot();
    await fireClickAndMouseEvents(sendTestBtn);
    await waitFor(() => {
      expect(getByAltText('Test Alert received')).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it('should show error details for a failing test-alert', async () => {
    const deliveryResponseMsg = 'This destination failed to pass the test';
    const deliveryResponse = buildDeliveryResponse({
      message: deliveryResponseMsg,
      statusCode: 400,
      success: false,
    });
    const mocks = [
      mockSendTestAlert({
        variables: {
          input: buildSendTestAlertInput({ outputIds: [destination.outputId] }),
        },
        data: { sendTestAlert: [deliveryResponse] },
      }),
    ];
    const { getByText, container, queryByAltText } = render(
      <Wizard initialData={{ destination }}>
        <Wizard.Step title="title">
          <WizardPanel>
            <DestinationTestPanel />
          </WizardPanel>
        </Wizard.Step>
      </Wizard>,
      { mocks }
    );

    expect(getByText('Everything looks good!')).toBeInTheDocument();
    const sendTestBtn = getByText('Send Test Alert');
    await fireClickAndMouseEvents(sendTestBtn);
    await waitFor(() => {
      expect(
        getByText(
          'Something went wrong and the destination you have configured did not receive the test alert. Please update your destination settings and try again.'
        )
      ).toBeInTheDocument();
      expect(getByText('Dispatched at')).toBeInTheDocument();

      expect(getByText('Status Code')).toBeInTheDocument();
      expect(getByText(deliveryResponseMsg)).toBeInTheDocument();
      expect(getByText('400')).toBeInTheDocument();
    });

    expect(queryByAltText('Test Alert received')).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
