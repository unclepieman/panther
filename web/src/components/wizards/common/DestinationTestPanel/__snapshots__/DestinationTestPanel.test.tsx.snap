// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationTestPanel should show a success message for successful test-alert delivery 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

.panther-17 {
  background-color: #23344e;
  border-radius: 4px;
  padding: 24px;
  margin-bottom: 24px;
  width: 100%;
  position: relative;
}

.panther-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding-top: 10px;
  margin-bottom: 60px;
  z-index: 2;
}

.panther-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 2;
  opacity: 1;
}

.panther-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 25px;
  height: 25px;
  font-size: 0.75rem;
  font-weight: 700;
  border-radius: 99999px;
  border: 1px solid;
  border-color: #bdbdbd;
  background-color: transparent;
}

.panther-1 {
  font-size: 0.875rem;
  margin-left: 8px;
}

.panther-16 {
  padding-top: 12px;
}



.panther-14 {
  will-change: opacity;
  -webkit-animation: animation-0 300ms 0ms ease-in-out both;
  animation: animation-0 300ms 0ms ease-in-out both;
}

.panther-13 {
  max-width: 700px;
  margin-left: auto;
  margin-right: auto;
}

.panther-6 {
  margin-bottom: 40px;
  text-align: center;
}

.panther-4 {
  font-weight: 500;
  font-size: 1.25rem;
}

.panther-5 {
  font-size: 0.875rem;
  margin-top: 8px;
  color: #bdbdbd;
}

.panther-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 24px;
  margin-bottom: 24px;
}

.panther-12>*:not(:last-child) {
  margin-bottom: 24px;
}

.panther-7 {
  max-width: 100%;
}

.panther-8 {
  margin-bottom: 20px;
}

.panther-9 {
  width: auto;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  height: 46px;
  font-size: 0.938rem;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  border: 1px solid;
  border-color: #0081c5;
  background-color: #0081c5;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-9:hover {
  background-color: #0a8bcf;
  border-color: #0a8bcf;
}

.panther-9:focus {
  background-color: #0a8bcf;
  border-color: #55d6ff;
}

.panther-9:active,
.panther-9[data-active=true] {
  background-color: #0077bb;
  border-color: #0077bb;
}

.panther-11 {
  font-size: 0.625rem;
  font-weight: 700;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #dedede;
}

<div>
  <header
    id="main-header"
  />
  <article
    class="panther-17"
  >
    <ul
      class="panther-3"
    >
      <li
        class="panther-2"
      >
        <div
          class="panther-0"
        >
          1
        </div>
        <p
          class="panther-1"
        >
          title
        </p>
      </li>
    </ul>
    <div
      class="panther-16"
    >
      <section
        class="panther-10"
      >
        <div
          class="panther-14"
        >
          <div
            class="panther-13"
          >
            <header
              class="panther-6"
            >
              <h1
                class="panther-4"
              >
                Everything looks good!
              </h1>
              <p
                class="panther-5"
              >
                Your destination was successfully added and you will receive alerts based on your configuration. You can always edit or delete this destination from the destinations page
              </p>
            </header>
            <div
              class="panther-12"
            >
              <img
                alt="Success"
                class="panther-7"
                height="120"
                src="test-file-stub"
                width="120"
              />
              <p
                class="panther-8"
              >
                Do you want to try it out by sending a test Alert?
              </p>
              <div
                class="panther-10"
              >
                <button
                  class="panther-9"
                  type="button"
                >
                  Send Test Alert
                </button>
              </div>
              <a
                class="panther-11"
                href="/integrations/destinations/"
              >
                Finish Setup
              </a>
            </div>
          </div>
        </div>
      </section>
    </div>
  </article>
  <footer
    id="footer"
  />
</div>
`;

exports[`DestinationTestPanel should show a success message for successful test-alert delivery 2`] = `
@keyframes animation-0 {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

.panther-17 {
  background-color: #23344e;
  border-radius: 4px;
  padding: 24px;
  margin-bottom: 24px;
  width: 100%;
  position: relative;
}

.panther-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding-top: 10px;
  margin-bottom: 60px;
  z-index: 2;
}

.panther-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 2;
  opacity: 1;
}

.panther-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 25px;
  height: 25px;
  font-size: 0.75rem;
  font-weight: 700;
  border-radius: 99999px;
  border: 1px solid;
  border-color: #bdbdbd;
  background-color: transparent;
}

.panther-1 {
  font-size: 0.875rem;
  margin-left: 8px;
}

.panther-16 {
  padding-top: 12px;
}



.panther-14 {
  will-change: opacity;
  -webkit-animation: animation-0 300ms 0ms ease-in-out both;
  animation: animation-0 300ms 0ms ease-in-out both;
}

.panther-13 {
  max-width: 700px;
  margin-left: auto;
  margin-right: auto;
}

.panther-6 {
  margin-bottom: 40px;
  text-align: center;
}

.panther-4 {
  font-weight: 500;
  font-size: 1.25rem;
}

.panther-5 {
  font-size: 0.875rem;
  margin-top: 8px;
  color: #bdbdbd;
}

.panther-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 24px;
  margin-bottom: 24px;
}

.panther-12>*:not(:last-child) {
  margin-bottom: 24px;
}

.panther-7 {
  max-width: 100%;
}

.panther-9 {
  width: auto;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  height: 46px;
  font-size: 0.938rem;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  border: 1px solid;
  border-color: #0081c5;
  background-color: #0081c5;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-9:hover {
  background-color: #0a8bcf;
  border-color: #0a8bcf;
}

.panther-9:focus {
  background-color: #0a8bcf;
  border-color: #55d6ff;
}

.panther-9:active,
.panther-9[data-active=true] {
  background-color: #0077bb;
  border-color: #0077bb;
}



.panther-10>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-11 {
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
  -webkit-transition: all 0.1s linear;
  transition: all 0.1s linear;
  font-size: 0.625rem;
  font-weight: 700;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #dedede;
}

<div>
  <header
    id="main-header"
  />
  <article
    class="panther-17"
  >
    <ul
      class="panther-3"
    >
      <li
        class="panther-2"
      >
        <div
          class="panther-0"
        >
          1
        </div>
        <p
          class="panther-1"
        >
          title
        </p>
      </li>
    </ul>
    <div
      class="panther-16"
    >
      <section
        class="panther-8"
      >
        <div
          class="panther-14"
        >
          <div
            class="panther-13"
          >
            <header
              class="panther-6"
            >
              <h1
                class="panther-4"
              >
                Testing your Destination
              </h1>
              <p
                class="panther-5"
              >
                Everything worked as planned and your destination received the triggered alert. You can always send additional test alerts from the destinations page.
              </p>
            </header>
            <div
              class="panther-12"
            >
              <img
                alt="Test Alert received"
                class="panther-7"
                height="120"
                src="test-file-stub"
                width="120"
              />
              <p
                class="panther-8"
              >
                Signed, sealed, and delivered. You are good to go!
              </p>
              <a
                class="panther-10"
                href="/integrations/destinations/"
              >
                <span
                  class="panther-9"
                  type="button"
                >
                  Finish Setup
                </span>
              </a>
              <button
                class="panther-11"
                type="button"
              >
                Add Another
              </button>
            </div>
          </div>
        </div>
      </section>
    </div>
  </article>
  <footer
    id="footer"
  />
</div>
`;

exports[`DestinationTestPanel should show error details for a failing test-alert 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

.panther-28 {
  background-color: #23344e;
  border-radius: 4px;
  padding: 24px;
  margin-bottom: 24px;
  width: 100%;
  position: relative;
}

.panther-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding-top: 10px;
  margin-bottom: 60px;
  z-index: 2;
}

.panther-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 2;
  opacity: 1;
}

.panther-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 25px;
  height: 25px;
  font-size: 0.75rem;
  font-weight: 700;
  border-radius: 99999px;
  border: 1px solid;
  border-color: #bdbdbd;
  background-color: transparent;
}

.panther-1 {
  font-size: 0.875rem;
  margin-left: 8px;
}

.panther-27 {
  padding-top: 12px;
}



.panther-25 {
  will-change: opacity;
  -webkit-animation: animation-0 300ms 0ms ease-in-out both;
  animation: animation-0 300ms 0ms ease-in-out both;
}

.panther-24 {
  max-width: 700px;
  margin-left: auto;
  margin-right: auto;
}

.panther-6 {
  margin-bottom: 40px;
  text-align: center;
}

.panther-4 {
  font-weight: 500;
  font-size: 1.25rem;
}

.panther-5 {
  font-size: 0.875rem;
  margin-top: 8px;
  color: #bdbdbd;
}

.panther-23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 24px;
  margin-bottom: 24px;
}

.panther-23>*:not(:last-child) {
  margin-bottom: 24px;
}

.panther-20 {
  width: auto;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  height: 46px;
  font-size: 0.938rem;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  border: 1px solid;
  border-color: #0081c5;
  background-color: #0081c5;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-20:hover {
  background-color: #0a8bcf;
  border-color: #0a8bcf;
}

.panther-20:focus {
  background-color: #0a8bcf;
  border-color: #55d6ff;
}

.panther-20:active,
.panther-20[data-active=true] {
  background-color: #0077bb;
  border-color: #0077bb;
}

.panther-22 {
  font-size: 0.625rem;
  font-weight: 700;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #dedede;
}



.panther-21>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-18 {
  background-color: #87374b;
  border-radius: 4px;
  padding: 24px;
}

.panther-17 {
  display: grid;
  grid-gap: 16px;
  grid-template-columns: max-content 1fr;
  word-break: break-word;
  font-size: 0.875rem;
  font-weight: 500;
}

.panther-7 {
  margin-top: auto;
  margin-bottom: auto;
}

.panther-8 {
  font-weight: 700;
}

<div>
  <header
    id="main-header"
  />
  <article
    class="panther-28"
  >
    <ul
      class="panther-3"
    >
      <li
        class="panther-2"
      >
        <div
          class="panther-0"
        >
          1
        </div>
        <p
          class="panther-1"
        >
          title
        </p>
      </li>
    </ul>
    <div
      class="panther-27"
    >
      <section
        class="panther-19"
      >
        <div
          class="panther-25"
        >
          <div
            class="panther-24"
          >
            <header
              class="panther-6"
            >
              <h1
                class="panther-4"
              >
                Testing your Destination
              </h1>
              <p
                class="panther-5"
              >
                Something went wrong and the destination you have configured did not receive the test alert. Please update your destination settings and try again.
              </p>
            </header>
            <div
              class="panther-23"
            >
              <div
                class="panther-18"
              >
                <dl
                  class="panther-17"
                >
                  <dt
                    class="panther-7"
                  >
                    Dispatched at
                  </dt>
                  <dd
                    class="panther-8"
                  >
                    2020-09-25T00:14:42.514Z
                  </dd>
                  <dt
                    class="panther-7"
                  >
                    Message
                  </dt>
                  <dd
                    class="panther-8"
                  >
                    This destination failed to pass the test
                  </dd>
                  <dt
                    class="panther-7"
                  >
                    Output ID
                  </dt>
                  <dd
                    class="panther-8"
                  >
                    bb9f4174-594c-4dc0-9308-f4c28c0e29eb
                  </dd>
                  <dt
                    class="panther-7"
                  >
                    Status Code
                  </dt>
                  <dd
                    class="panther-8"
                  >
                    400
                  </dd>
                  <dt
                    class="panther-7"
                  >
                    Success
                  </dt>
                  <dd
                    class="panther-8"
                  >
                    false
                  </dd>
                </dl>
              </div>
              <p
                class="panther-19"
              >
                If you don
                '
                t feel like it right now, you can always change the configuration later
              </p>
              <a
                class="panther-21"
                href="/integrations/destinations/8c0eb672-b7bb-4ef0-9d96-a2bc1abe94d7/edit/"
              >
                <span
                  class="panther-20"
                  type="button"
                >
                  Back to Configuration
                </span>
              </a>
              <a
                class="panther-22"
                href="/integrations/destinations/"
              >
                Skip Testing
              </a>
            </div>
          </div>
        </div>
      </section>
    </div>
  </article>
  <footer
    id="footer"
  />
</div>
`;
